### PR TITLE
Fix a missing await when in the spaces summary

### PR DESCRIPTION
**Pull request by clokep**
* Adds a couple of unit tests.
* Fixes a missing `await` in the spaces summary API. (This was introduced in #10109.)

### CODE REVIEW

File: changelog.d/10208.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix a bug introduced in v1.35.1 where an `allow` key of a `m.room.join_rules` event could be applied for incorrect room versions and configurations.
```
Comment (clokep) [inline changelog.d/10208.bugfix:1]: I'm not sure this explains it well enough, but not awaiting there it means that you could have a room with `allow` set in the join rules, but a join rule of *not* `restricted` and those allow rules would still be applied.

Realistically I don't think this is user visible unless someone was messing with join rules in a weird way, although it means you could slightly leak data if you did this.

Comment (anoadragon453) [reply changelog.d/10208.bugfix:1]: If this was more user-visible, I'd probably suggest rephrasing "an `allow` key of a `m.room.join_rules` event could be applied" with something more user-digestible, but as it's not it's likely more useful to be clear about what exactly the missing await affected. In which case, to a developer, this is very clear :slightly_smiling_face:

**Review by anoadragon453 — APPROVED** [review]
lgtm, and thanks for adding a test. It was also very useful to see the test built up over a series of commits :slightly_smiling_face:

**Comment by clokep** [comment]
Thanks! 🎉

**Pull request merged**